onnxruntime/core/framework/execution_frame.cc

```diff
@@ -729,8 +733,26 @@ Status ExecutionFrame::AllocateAsPerAllocationPlan(OrtValue& ort_value, int ort_
     // Model load should have failed so this should be unreachable
     ORT_THROW("SparseTensor is not supported in this build.");
 #endif
-  } else if (ml_type->IsTensorSequenceType()) {
-    return AllocateTensorSequence(ort_value);
+  } else if (ml_type->IsTensorSequenceType() || utils::IsOptionalSeqTensor(ml_type)) {
+    AllocKind alloc_kind = per_alloc_plan.alloc_kind;
+
+    if (alloc_kind == AllocKind::kReuse) {
+      int reuse_mlvalue_index = per_alloc_plan.reused_buffer;
+
+      // In case OrtRunOptions.only_execute_path_to_fetches == true, it is possible that 'reuse_value'
+      // is not allocated (its upstream op is not executed due to the option).
+      // In this case we need to allocate 'reuse_value' and then let 'ort_value' to reuse it.
+      OrtValue& reuse_value = GetMutableMLValue(reuse_mlvalue_index);
+      if (!reuse_value.IsAllocated()) {
+        ORT_RETURN_IF_ERROR(AllocateAsPerAllocationPlan(reuse_value, reuse_mlvalue_index, shape));
+      }
+
```

Comment:
nit: could do this via a helper as it's a repeated pattern (comment is cut-and-pasted from line 699)

Reply:
Done